polars-lazy fails to compile with `super::get_glob_start_idx` error

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

This isn't failing in my code, the compiler is failing in `polars-lazy-0.41.0/src/scan/parquet.rs:8:5`


### Log output

```shell
🕙 03:03:22 ❯ cargo build
   Compiling polars-lazy v0.41.0
error[E0432]: unresolved import `super::get_glob_start_idx`
 --> /Users/nick/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-lazy-0.41.0/src/scan/parquet.rs:8:5
  |
8 | use super::get_glob_start_idx;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^ no `get_glob_start_idx` in `scan`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `polars-lazy` (lib) due to 1 previous error
```


### Issue description

polars-lazy is failing to compile. It looks like this code was just added:

https://github.com/pola-rs/polars/commit/7f5412a1f01e7ae95b98fb0a5412779ecea47c22

I've tried `cargo clean && cargo build` as well as removing `Cargo.lock`, but the error persists.

### Expected behavior

Successful compile.

### Installed versions

<details>

polars = { version = "0.*", features = ["parquet", "lazy", "dtype-categorical", "dtype-i16"] }
polars 0.41.0
rustc 1.78.0 (9b00956e5 2024-04-29)
</details>

--- a/crates/polars-lazy/src/scan/mod.rs
+++ b/crates/polars-lazy/src/scan/mod.rs
@@ -8,6 +8,3 @@ pub(super) mod ipc;
 pub(super) mod ndjson;
 #[cfg(feature = "parquet")]
 pub(super) mod parquet;
-
-#[cfg(feature = "json")]
-use file_list_reader::*;

--- a/crates/polars-lazy/src/scan/ndjson.rs
+++ b/crates/polars-lazy/src/scan/ndjson.rs
@@ -6,8 +6,8 @@ use polars_io::RowIndex;
 use polars_plan::plans::{DslPlan, FileScan};
 use polars_plan::prelude::{FileScanOptions, NDJsonReadOptions};
 
-use super::*;
 use crate::prelude::LazyFrame;
+use crate::scan::file_list_reader::LazyFileListReader;
 
 #[derive(Clone)]
 pub struct LazyJsonLineReader {

--- a/crates/polars-lazy/src/scan/parquet.rs
+++ b/crates/polars-lazy/src/scan/parquet.rs
@@ -5,8 +5,8 @@ use polars_io::cloud::CloudOptions;
 use polars_io::parquet::read::ParallelStrategy;
 use polars_io::{HiveOptions, RowIndex};
 
-use super::get_glob_start_idx;
 use crate::prelude::*;
+use crate::scan::file_list_reader::get_glob_start_idx;
 
 #[derive(Clone)]
 pub struct ScanArgsParquet {

